detox/android/detox/src/main/java/com/wix/detox/espresso/utils/AgnosticPoint2D.kt

```diff
@@ -0,0 +1,36 @@
+package com.wix.detox.espresso.utils
+
+import com.wix.detox.common.DetoxErrors
+import com.wix.detox.espresso.common.annot.MOTION_DIR_DOWN
+import com.wix.detox.espresso.common.annot.MOTION_DIR_LEFT
+import com.wix.detox.espresso.common.annot.MOTION_DIR_RIGHT
+import com.wix.detox.espresso.common.annot.MOTION_DIR_UP
+
+data class FloatingPoint(val x: Float, val y: Float)
+
+data class AgnosticPoint2D(val primary: Double, val secondary: Double) {
```

Comment:
So if I understand correctly, this really represents motion, or a _translation_, along 2 axes - one being the primary and the other, secondary. Since the translation is given in ratio rather than actual points, it is agnostic.
Wdyt about things like `AgnosticVector2D` for a class name, then? maybe `ProportionateVector2D`? And `primaryAxis`, `secondaryAxis` for params?

Reply:
Not relevant anymore — I've rewritten the implementation to use vectors.